### DISCUSSION TITLE
Supabase storage owner column empty

### DISCUSSION THREAD

**Discussion by bombillazo** [supabase/supabase#15541] [General]
Hey, I am inserting files in my backend using the service-role key, and in the `objects` table I see the owner is empty. What is this column used for?

**Comment by GaryAustin1** [comment]
So you are posting elsewhere and not sure if that resolved it for you or not.  Owner is the id (sub claim in the jwt) of the user creating the file.  In the case of the UI or service_role that is null.

**Reply by bombillazo** [reply]
How can i pass the id I want to use when the service role is used?

**Reply by GaryAustin1** [reply]
You can't.  

If you are using service role though, that means you are on a server or edge function.   You can mint a Supabase compatible jwt with the user id in the sub and the other claims using the jwt secret key and a jwt library.   https://supabase.com/docs/learn/auth-deep-dive/auth-deep-dive-jwts#jwts-in-supabase has info on the jwt.

You will need to meet any policy you have if you use the user claims.

Edit... I guess you could also do a database call to modify the owner column in the storage.objects table in a 2nd call if you wanted.

**Reply by bombillazo** [reply]
Crap... This shouldn't be this hard... Ok i think ill go the auth jwt route... Thanks

**Reply by armartinez** [reply]
@GaryAustin1 I'm trying to update the file ownership when uploading a file for a user as an admin using RBAC by making a database call to the owner column in the storage.objects, but it's seems only the postgres role can do it...

**Reply by GaryAustin1** [reply]
What error are you getting?
How are you doing this update?
Do you meet RLS for your RBAC both with the current user id and the new user id?
Storage.objects should be accessible to authenticated users if RLS is met.

**Reply by armartinez** [reply]
@GaryAustin1 it's strange because I get an empty error object on the js client side. If I try to do it directly using a SQL Query it looks like RLS steps in but I think the policies are correct. I have to move on to other things right now but I'll do more testing and update my comment here. Thanks for the quick reply!

**Reply by armartinez** [reply]
The problem was that I didn't have the storage schema available on the API, after adding it it works, in case somebody else stumbles into this.
